### PR TITLE
Movie Detail Screen feature.

### PR DESCRIPTION
**Pull request by SebasAMont**
Implementation of movie detail screen, fetch movie detail, UI fixes and tests. 

https://user-images.githubusercontent.com/109153713/199606452-ec1e5677-3dc3-47c3-976f-94c1a3c45054.mp4

image (1)

image

### CODE REVIEW

File: lib/presentation/widget/visibility/visibility_list_image.dart [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                        movieDetailList[index]!.logoPath,
+                        Dimensions.smallPosterSize,
+                      )
+                    : const Text('')
+                : (movieDetailList[index].posterPath != null)
+                    ? showImgIfExists(
+                        movieDetailList[index]!.posterPath,
+                        Dimensions.smallPosterSize,
+                      )
+                    : const Text(''),
```
Comment (MarcosCGlobant) [inline lib/presentation/widget/visibility/visibility_list_image.dart]: Why do you need this empty string Texts?

**Review by FranciscoGaraventa — CHANGES_REQUESTED** [review]
Validate widget abstraction, all of these classes could be included in one widget

- VisibilityListAttributesProductionCompany
- VisibilityListAttributesProductionCountry
- VisibilityListAttributesSpokenLanguage
- VisibilityListImage (visibility_list_image / visibility_list_image_collection)

Changed text style class, this could be as static inside constants.

File: lib/presentation/widget/visibility/visibility_list_image.dart (hunk 1 above)
Comment (FacuCristobalGlobant) [reply lib/presentation/widget/visibility/visibility_list_image.dart]: We removed that file and use visibility_list_of_strings instead, we no longer use that portion of code.

**Review by MarcosCGlobant — APPROVED** [review]

File: lib/data/model/production_companies_model.dart [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+    );
+  }
+
+  @override
+  String toString() {
+    if (logoPath != null) {
+      return '$name#$logoPath';
+    } else {
+      return '$name';
+    }
```
Comment (FranciscoGaraventa) [inline lib/data/model/production_companies_model.dart]: Here you can use a ternary condition

File: lib/presentation/view/movie_detail_screen.dart [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+
+  @override
+  Widget build(BuildContext context) {
+    return SafeArea(
+      child: Scaffold(
+        backgroundColor: Palette.background,
+        body: CustomScrollView(
+          slivers: <Widget>[
+            SliverAppBar(
+              expandedHeight: 200,
```
Comment (FranciscoGaraventa) [inline lib/presentation/view/movie_detail_screen.dart]: Add expandedHeight to consts

Comment (marcosvonstremayr) [reply lib/presentation/view/movie_detail_screen.dart]: done

File: lib/data/model/production_companies_model.dart (hunk 2 above)
Comment (marcosvonstremayr) [reply lib/data/model/production_companies_model.dart]: done

**Review by FranciscoGaraventa — APPROVED** [review]
LG

**Pull request merged**